x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/rerank/HuggingFaceRerankTaskSettings.java

```diff
@@ -0,0 +1,156 @@
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalBoolean;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalPositiveInteger;
+
+public class HuggingFaceRerankTaskSettings implements TaskSettings {
+
+    public static final String NAME = "hugging_face_rerank_task_settings";
+    public static final String RETURN_TEXT = "return_text";
+    public static final String TOP_N_DOCS_ONLY = "top_n";
+
+    static final HuggingFaceRerankTaskSettings EMPTY_SETTINGS = new HuggingFaceRerankTaskSettings(null, null);
+
+    public static HuggingFaceRerankTaskSettings fromMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+
+        if (map == null || map.isEmpty()) {
+            return EMPTY_SETTINGS;
+        }
+
+        Boolean returnDocuments = extractOptionalBoolean(map, RETURN_TEXT, validationException);
+        Integer topNDocumentsOnly = extractOptionalPositiveInteger(
+            map,
+            TOP_N_DOCS_ONLY,
+            ModelConfigurations.TASK_SETTINGS,
+            validationException
+        );
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+
+        return of(topNDocumentsOnly, returnDocuments);
+    }
+
+    /**
+     * Creates a new {@link HuggingFaceRerankTaskSettings}
+     * by preferring non-null fields from the request settings over the original settings.
+     *
+     * @param originalSettings    the settings stored as part of the inference entity configuration
+     * @param requestTaskSettings the settings passed in within the task_settings field of the request
+     * @return a constructed {@link HuggingFaceRerankTaskSettings}
+     */
+    public static HuggingFaceRerankTaskSettings of(
+        HuggingFaceRerankTaskSettings originalSettings,
+        HuggingFaceRerankTaskSettings requestTaskSettings
+    ) {
+        return new HuggingFaceRerankTaskSettings(
+            requestTaskSettings.getTopNDocumentsOnly() != null
+                ? requestTaskSettings.getTopNDocumentsOnly()
+                : originalSettings.getTopNDocumentsOnly(),
+            requestTaskSettings.getReturnDocuments() != null
+                ? requestTaskSettings.getReturnDocuments()
+                : originalSettings.getReturnDocuments()
+        );
+    }
+
+    public static HuggingFaceRerankTaskSettings of(Integer topNDocumentsOnly, Boolean returnDocuments) {
+        return new HuggingFaceRerankTaskSettings(topNDocumentsOnly, returnDocuments);
+    }
+
+    private final Integer topNDocumentsOnly;
+    private final Boolean returnDocuments;
+
+    public HuggingFaceRerankTaskSettings(StreamInput in) throws IOException {
+        this(in.readOptionalInt(), in.readOptionalBoolean());
```

Comment:
Instead of `readOptionalInt` let's use `readOptionalVInt` and `writeOptionalVInt`.